src/tactics/FStar.Tactics.Hooks.fs

```diff
@@ -115,6 +119,43 @@ let by_tactic_interp (pol:pol) (e:Env.env) (t:term) : tres =
             Simplified (assertion, [])
         end
 
+    // rewrite_with_tactic marker
+    | Tm_fvar fv, [(tactic, None); (typ, Some (Implicit false)); (tm, None)]
+            when S.fv_eq_lid fv PC.rewrite_by_tactic_lid ->
+
+        // Create a new uvar that must be equal to the initial term
+        let uvtm, _, g_imp = Env.new_implicit_var_aux "rewrite_with_tactic RHS" tm.pos e typ Allow_untyped None in
+
+        let u = e.universe_of e typ in
+        // eq2 is squashed already, so it's in Type0
+        let goal = U.mk_squash U_zero (U.mk_eq2 u typ tm uvtm) in
+        let gs, _ = run_tactic_on_typ tactic.pos tm.pos tactic e goal in
+
+        // Ensure that rewriting did not leave goals
+        List.iter (fun g ->
```

Comment:
As far as I understand, a user tactic can always drop all the goals using `FStar.Tactics.Builtins.set_goals`, so is this check adding any insurance? Why can't you just drop this loop, resolve_implicits, and then report if some remain unresolved?

Reply:
That does sound easier. I'm just a bit confused about what goal I should run force_trivial_guard on if not the list of remaining goals.